spec/spec_helper_acceptance.rb

```diff
@@ -1,20 +1,12 @@
 #! /usr/bin/env ruby -S rspec
 require 'beaker-rspec'
+require 'beaker/puppet_install_helper'
 
 UNSUPPORTED_PLATFORMS = []
 
-unless ENV['RS_PROVISION'] == 'no' or ENV['BEAKER_provision'] == 'no'
-  if hosts.first.is_pe?
-    install_pe
-    on hosts, 'mkdir -p /etc/puppetlabs/facter/facts.d'
-  else
-    install_puppet
-    on hosts, 'mkdir -p /etc/facter/facts.d'
-    on hosts, '/bin/touch /etc/puppet/hiera.yaml'
-  end
-  hosts.each do |host|
-    on host, "mkdir -p #{host['distmoduledir']}"
-  end
+hosts.each do |_host|
+  # Install Puppet
+  run_puppet_install_helper
```

Comment:
I think `run_puppet_install_helper` runs at the top level and knows to install on each host. https://github.com/puppetlabs/beaker-puppet_install_helper/blob/master/lib/beaker/puppet_install_helper.rb#L18-L25